lib/tbot/service_spiffe_svid_output.go

```diff
@@ -134,8 +197,35 @@ func (s *SPIFFESVIDOutputService) generate(ctx context.Context) error {
 		s.botCfg.CertificateTTL,
 	)
 	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+	return res, privateKey, nil
+}
+
+func (s *SPIFFESVIDOutputService) render(
+	ctx context.Context,
+	bundleSet *spiffe.BundleSet,
+	res *machineidv1pb.SignX509SVIDsResponse,
+	privateKey *rsa.PrivateKey,
+) error {
+	ctx, span := tracer.Start(
+		ctx,
+		"SPIFFESVIDOutputService/render",
+	)
+	defer span.End()
+	s.log.InfoContext(ctx, "Rendering output")
+
+	// Check the ACLs. We can't fix them, but we can warn if they're
```

Comment:
Why was this moved down to the `render()` function?

Reply:
I think the diff really looks a bit odd here since the PR introduced the render function. It makes more sense for this to be in render since it actually touches/writes to the destination.

Reply:
(Admittedly most other outputs make these verification calls during `generate()` but it's all pretty arbitrary. The two could just as well be one long function.)